src/lib/PropTypes.js

```diff
@@ -1,24 +1,38 @@
 import uniq from 'lodash/uniq';
 
+const varTypes = {
+  string: String,
+  number: Number,
+  boolean: Boolean,
+};
+
 export function Enum(values = [], obj = {}) {
   let type;
-  const choices = values;
-
-  if (obj.type && Array.isArray(obj.type)) {
-    if (obj.type.includes(Boolean)) choices.push('');
-    type = uniq([...obj.type, String]);
-  } else if (obj.type) {
-    if (obj.type === Boolean) choices.push('');
-    type = [String, obj.type];
+  const choices = [...values];
+
+  if (Array.isArray(obj.choices)) {
+    choices.push(...obj.choices);
+  }
+
+  const types = uniq(values.map(value => varTypes[typeof value]));
```

Comment:
I think you can remove `varTypes` and use `value.constructor`

Reply:
Right